core/arch/arm/plat-nuvoton/vendor_props.c

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2022-2023 Nuvoton Ltd.
+ */
+
+#include <kernel/tee_common_otp.h>
+#include <kernel/tee_ta_manager.h>
+#include <kernel/user_access.h>
+#include <tee/tee_cryp_utl.h>
+#include <tee/tee_svc.h>
+#include <user_ta_header.h>
+#include <util.h>
+
+/*
+ * The data to hash is 48 bytes made up of:
+ * - 16 bytes: the UUID of the calling TA.
+ * - 32 bytes: the hardware device ID
+ * The resulting endorsement seed is 32 bytes.
+ *
+ * The output buffer is the "binary" struct defined in
+ * the "prop_value" union and therefore comprises:
+ * -  4 bytes: the size of the binary value data (32)
+ * - 32 bytes: the binary value data (endorsement seed)
+ *
+ * Note that this code assumes an endorsement seed
+ * size == device ID size for convenience.
+ */
+#define TA_ENDORSEMENT_SEED_SIZE	TEE_SHA256_HASH_SIZE
+
+static TEE_Result get_prop_endorsement(struct ts_session *sess, void *buf,
+				       size_t *blen)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	uint8_t data[sizeof(TEE_UUID) + TA_ENDORSEMENT_SEED_SIZE] = { };
+	uint32_t bin[1 + TA_ENDORSEMENT_SEED_SIZE / sizeof(uint32_t)] = { };
+	uint32_t *bin_len = bin;
+
+	if (*blen < sizeof(bin)) {
+		*blen = sizeof(bin);
+		return TEE_ERROR_SHORT_BUFFER;
+	}
+
+	*blen = sizeof(bin);
+
+	memcpy(data, &sess->ctx->uuid, sizeof(TEE_UUID));
+
+	if (tee_otp_get_die_id(&data[sizeof(TEE_UUID)],
+			       TA_ENDORSEMENT_SEED_SIZE))
+		return TEE_ERROR_BAD_STATE;
+
+	bin[0] = TA_ENDORSEMENT_SEED_SIZE;
+	res = tee_hash_createdigest(TEE_ALG_SHA256, data, sizeof(data),
+				    (uint8_t *)&bin[1], TA_ENDORSEMENT_SEED_SIZE);
```

Comment:
Needs a line break before last argument otherwise checkpatch will complain.
Refer to https://optee.readthedocs.io/en/latest/general/coding_standards.html#running-checkpatch for how to run checkpatch on your commits before posting them.

Reply:
Done